external-plugins/rehearse/plugin/main.go

```diff
@@ -0,0 +1,189 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"k8s.io/client-go/rest"
+	v1 "k8s.io/test-infra/prow/client/clientset/versioned/typed/prowjobs/v1"
+	"net/http"
+
+	"k8s.io/client-go/tools/clientcmd"
+
+	"k8s.io/test-infra/prow/interrupts"
+
+	"os"
+	"time"
+
+	"k8s.io/test-infra/prow/git/v2"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/test-infra/pkg/flagutil"
+	"k8s.io/test-infra/prow/config/secret"
+	prowflagutil "k8s.io/test-infra/prow/flagutil"
+	"kubevirt.io/project-infra/external-plugins/rehearse/events_server"
+)
+
+type options struct {
+	dryRun         bool
+	hmacSecretFile string
+	endpoint       string
+	port           int
+	prowConfigPath string
+	jobsConfigBase string
+	kubeconfig     string
+	jobsNs         string
+	alwaysRun      bool
+	cacheDir       string
+	github         prowflagutil.GitHubOptions
+}
+
+func (o *options) validate() {
```

Comment:
What about using third parties for arguments handling that implementes the mandatory attribute ?

Reply:
I inherit some flags from Prow's module and I need flag to register those. Otherwise, I would have used spf13/cobra.

Reply:
Fair enough